File: Programmers/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    int size = number.length();
    int max_idx = 0; // 비교용
    
    // 변수 size 길이의 문자열안에 k만큼 써야함
    for (int i = 1; i <= size - k; ++i) {
        int max_number = -1; // 탐색 전 초기화
        // max_idx부터 K만큼 큰 수 탐색
        // 처음에는 0
        for (int j = max_idx ; j < k + i; ++j) {
            if (max_number < number[j] - '0') {
                max_number = number[j] - '0';
                max_idx = j; // 스티트 구분을 위해
            }
        }
        answer += (max_number + '0');
        max_idx++;
    }
    return answer;
}

File: Programmers/큰 수 만들기.py
# level 2
def solution(number, k):
    answer = ''
    stack = []
    
    for num in number:
        # 반복
        # 추가해야할 k가 있고 스택 top보다 큰 값을 만날때 까지
        while k > 0 and stack and stack[-1] < num:
            stack.pop()
            k -= 1
        # 중단되고 해당 값 스택에 삽입
        stack.append(num)
    
    # 차례로 끝까지 그냥 선택해야하는 경우도 존재
    # 그럼 k는 for문이 끝나도 0이 아님
    if k > 0:
        stack = stack[:-1]
    answer = ''.join(stack)
    return answer